### ISSUE TITLE
Nested routes using NuxtPage or NuxtChild don't get generated

### ISSUE THREAD

**Issue by tewshi** [harlan-zw/nuxt-simple-sitemap#23]
### Describe the bug

I have a directory structure
```
pages
   /blog.vue
   /blog
      /categories.vue
      /tags.vue
      /index.vue
```

pages/blog.vue has a `<NuxtPage />` which allows it serve as a root to the categories and tags.

But, the sitemap plugin doesn't scan through the child pages.

in code, it generates 
```typescript
{                                                                                                                                  17:45:23
  name: 'blog',
  path: '/blog',
  file: '[path-to-app]/pages/blog.vue',
  children: [
    {
      name: 'blog-categories',
      path: '/blog/categories',
      file: '[path-to-app]/pages/blog/categories.vue',
      children: []
    },
    {
      name: 'blog-tags',
      path: '/blog/tags',
      file: '[path-to-app]/pages/blog/tags.vue',
      children: []
    },
    {
      name: 'blog',
      path: '/blog/',
      file: '[path-to-app]/pages/blog/index.vue',
      children: []
    }
  ]
}
```

the children never gets generated.

### Reproduction

https://stackblitz.com/edit/nuxt-starter-sxgico?file=app.vue

### System / Nuxt Info

```Shell
------------------------------
- Operating System: `Linux`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.3.2`
- Nitro Version:    `2.3.2`
- Package Manager:  `npm@7.17.0`
- Builder:          `vite`
- User Config:      `modules`, `sitemap`
- Runtime Modules:  `nuxt-simple-sitemap@2.2.0`
- Build Modules:    `-`
------------------------------
```

**Comment by harlan-zw** [comment]
Thanks for the issue @tewshi 

I've pushed up a fix for this in v2.3.1, let me know if you have any further issues with it :)

**Comment by tewshi** [comment]
> Thanks for the issue @tewshi 
> 
> 
> 
> I've pushed up a fix for this in v2.3.1, let me know if you have any further issues with it :)

Great, thanks for the fix, I'll test again, if I find anything I'll file an issue. Well done

**Comment by Jeeying** [comment]
@harlan-zw thanks for the update. I've got a new problem.
If I have a multi-level directory structure, deeper directories cannot be generated.

```
pages
   /blog.vue                     --> generated
   /blog
      /categories.vue            --> generated
      /tags.vue                  --> generated
      /tags
         /new.vue                --> not generated
         /edit.vue               --> not generated
```

"nuxt-simple-sitemap": "^2.4.18"

**Comment by harlan-zw** [comment]
Thanks @Jeeying, looks like it wasn't scanning the children recurisvely, should be fixed in 2.4.19.

**Comment by Jeeying** [comment]
> Thanks @Jeeying, looks like it wasn't scanning the children recursively, should be fixed in 2.4.19.

yes, it's working!🎉🎉 thank you so much! that's very useful
